File: Kotlin/fatorial.kt
/*
 * N Fatorial (N!)
 *
 * by g_devo @2021
 *
 * Este script recebe um número inteiro n e retorna o fatorial de n.
 * Fatorial é o produto de um número multiplicado por seus antecessores até 1
 *
 */

fun main() {
    println("Digite o número: ")

    val inputNumber = readLine()!!.toInt()
    val fatorialResult = fatorial(inputNumber)

    println("O fatorial de $inputNumber é $fatorialResult")
}

fun fatorial(number: Int): Int {
    var fator = number
    var n = number
    while (n > 1) {
        n -= 1
        fator *= n
    }
    return fator
}

File: Python/fatorial.py
#!/usr/bin/env python
#
# N Fatorial (N!)
# by g_devo
#
# Este script recebe um número inteiro n e retorna o fatorial de n
#
# Fatorial é o produto de um número multiplicado por seus antecessores até 1
#


def fatorial(number: int) -> int:
    n = fator = number
    while n > 1:
        n -= 1
        fator *= n
    print(f'{number} fatorial é {fator}')
    return fator


if __name__ == '__main__':
    input_number = int(input('Digite o número: '))
    fatorial(input_number)
